web/packages/core/src/ruffle-player.ts

```diff
@@ -1994,24 +1994,31 @@ export class RufflePlayer extends HTMLElement {
                 });
             }
             this.hideSplashScreen();
+
             const div = document.createElement("div");
             div.id = "message_overlay";
+            const innerDiv = document.createElement("div");
+            innerDiv.className = "message";
+            innerDiv.innerHTML = textAsParagraphs("message-cant-embed");
 
+            const buttonDiv = document.createElement("div");
             const link = document.createElement("a");
             link.innerText = text("open-in-new-tab");
             link.onclick = () => openInNewTab(url);
+            buttonDiv.appendChild(link);
 
-            div.innerHTML = `<div class="message">
-                ${textAsParagraphs("message-cant-embed")}
-                <div></div>
-            </div>`;
-            div.lastChild!.appendChild(link);
+            innerDiv.appendChild(buttonDiv);
+            div.appendChild(innerDiv);
             this.container.prepend(div);
         } else {
             const error = new Error("Failed to fetch: " + this.swfUrl);
             if (!this.swfUrl!.protocol.includes("http")) {
                 error.ruffleIndexError = PanicError.FileProtocol;
-            } else if (window.location.origin === this.swfUrl!.origin) {
+            } else if (
+                window.location.origin === this.swfUrl!.origin ||
+                // The extension's internal player page is not restricted by CORS
+                window.location.protocol.includes("extension")
```

Comment:
Does this part work for Edge?

Reply:
Yes, Edge's extension protocol is called `extension://`. I've never seen a Chromium-based browser that used an extension protocol not including the word "extension" in its name.